config/initializers/will_paginate.rb

```diff
@@ -3,23 +3,42 @@
 
 module WillPaginate
   module ActionView
-    class BootstrapLinkRenderer < LinkRenderer
+    class PatternflyLinkRenderer < LinkRenderer
      protected
 
-      def html_container(html)
-        tag :div, tag(:ul, html, :class=>"pagination pull-right"), container_attributes
+      def container_attributes
+        super.except(:first_label, :last_label)
       end
 
-      def page_number(page)
-        tag :li, link(page, page, :rel => rel_value(page)), :class => ('active' if page == current_page)
+      def pagination
+        [:back, :pages, :forward]
       end
 
-      def previous_or_next_page(page, text, classname)
-        tag :li, link(text, page || '#'), :class => [classname[0..3], classname, ('disabled' unless page)].join(' ')
+      def back
+        tag :ul, first_page+previous_page, :class => "pagination pagination-pf-back"
+      end
+
+      def forward
+        tag :ul, next_page+last_page, :class => "pagination pagination-pf-forward"
+      end
+
+      def pages
+        current_page_input = tag(:input, "", :class => "pagination-pf-page", :type => "text", :value => current_page, :id => "pagination1-page")
+        current_page_label = tag(:label, _("Current Page"), :class => "sr-only", :for => "pagination1-page")
+        of_total_pages = tag(:span, _("of ") + tag(:span, total_pages, :class => "pagination-pf-pages"))
+        current_page_input + current_page_label + of_total_pages
       end
 
-      def gap
-        tag :li, link(super, '#'), :class => 'disabled'
+      def first_page
+        previous_or_next_page(current_page == 1 ? nil : 1, @options[:first_label], "first_page")
```

Comment:
is this a string that is used in the UI? if it is, it should be extracted (and others in this page).